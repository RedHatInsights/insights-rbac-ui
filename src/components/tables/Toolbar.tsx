import React, { RefObject } from 'react';
import PrimaryToolbar from '@redhat-cloud-services/frontend-components/PrimaryToolbar';
import { useIntl } from 'react-intl';
import messages from '../../Messages';
import { pickBy } from 'lodash';
import { calculateChecked, debouncedFetch, selectedRows } from '../../helpers/dataUtilities';
import { firstUpperCase } from '../../helpers/stringUtilities';
import { PER_PAGE_OPTIONS, calculateOffset, calculatePage, defaultSettings } from '../../helpers/pagination';

// Type definitions
interface PaginationData {
  limit?: number;
  offset?: number;
  count?: number;
}

export interface PaginationProps {
  toggleTemplate?: () => React.ReactElement;
  isCompact?: boolean;
}

interface FilterItem {
  key: string;
  value: string | number | string[] | unknown[];
  label?: string;
  selected?: boolean;
  placeholder?: string;
  type?: 'text' | 'checkbox' | 'group' | 'select';
  groups?: any[];
  items?: any[];
  innerRef?: RefObject<HTMLInputElement | HTMLTextAreaElement>;
}

interface BulkSelectItem {
  title: string;
  onClick: () => void;
}

interface BulkSelectConfig {
  count: number;
  items: BulkSelectItem[];
  checked: boolean | null;
  onSelect: (value: boolean) => void;
  id?: string;
  isDisabled?: boolean;
}

interface FilterConfig {
  onChange?: (event: any, value: any) => void;
  value?: any;
  items: any[];
}

interface ActiveFiltersConfig {
  filters: any[];
  onDelete: (event: any, deleted: any[], isAll?: boolean) => void;
}

interface ToolbarProps {
  isSelectable?: boolean;
  isRowSelectable?: (row: any) => boolean;
  checkedRows?: any[];
  setCheckedItems?: ((items: any[]) => void) | ((callback: (selected: any[]) => void) => void);
  isLoading?: boolean;
  data?: any[];
  titleSingular?: string;
  filterValue?: string | string[];
  setFilterValue?: (value: any) => void;
  filters?: FilterItem[];
  isFilterable?: boolean;
  onShowMore?: () => void;
  showMoreTitle?: string;
  onFilter?: (value: any) => void;
  onChange?: (event: any, value: any) => void;
  value?: any;
  pagination?: PaginationData;
  paginationProps?: PaginationProps;
  sortBy?: string;
  filterItems?: any[];
  filterPlaceholder?: string;
  fetchData: (params: any) => Promise<any> | void;
  toolbarButtons?: () => any[];
  hideFilterChips?: boolean;
  tableId?: string;
  textFilterRef?: RefObject<HTMLInputElement>;
  toolbarChildren?: () => React.ReactNode;
}

export const paginationBuilder = (
  pagination: PaginationData = defaultSettings,
  fetchData: (params: any) => Promise<any> | void,
  filterValue: string | string[] = '',
  sortBy = '',
  paginationProps?: PaginationProps,
) => ({
  ...paginationProps,
  itemCount: pagination.count,
  perPage: pagination.limit,
  page: calculatePage(pagination.limit, pagination.offset),
  onSetPage: (_event: any, page: number) => {
    fetchData({
      ...pagination,
      offset: calculateOffset(page, pagination.limit),
      name: filterValue,
      orderBy: sortBy,
    });
  },
  perPageOptions: PER_PAGE_OPTIONS,
  onPerPageSelect: (_event: any, perPage: number) => {
    fetchData({
      offset: 0,
      limit: perPage,
      name: filterValue,
      orderBy: sortBy,
    });
  },
});

export const bulkSelectBuilder = (
  isLoading: boolean,
  checkedRows: any[] = [],
  setCheckedItems: ((items: any[]) => void) | ((callback: (selected: any[]) => void) => void) = () => undefined,
  data: any[] = [],
  tableId?: string,
  isRowSelectable: (row: any) => boolean = () => true,
): BulkSelectConfig => {
  const intl = useIntl();

  const items: BulkSelectItem[] = [
    {
      title: intl.formatMessage(messages.selectNone),
      onClick: () => {
        if (typeof setCheckedItems === 'function') {
          (setCheckedItems as (callback: (selected: any[]) => void) => void)(() => []);
        }
      },
    },
  ];

  if (!isLoading && data && data.length > 0) {
    items.push({
      title: intl.formatMessage(messages.selectPage, {
        length: data.filter(isRowSelectable).length,
      }),
      onClick: () => {
        if (typeof setCheckedItems === 'function') {
          const selectedItems = selectedRows(data, true)(checkedRows);
          (setCheckedItems as (callback: (selected: any[]) => void) => void)(() => selectedItems);
        }
      },
    });
  }

  return {
    count: checkedRows.length,
    items,
    checked: calculateChecked(data, checkedRows, isRowSelectable),
    onSelect: (value: boolean) => {
      if (!isLoading && typeof setCheckedItems === 'function') {
        const selectedItems = selectedRows(data, value)(checkedRows);
        (setCheckedItems as (callback: (selected: any[]) => void) => void)(() => selectedItems);
      }
    },
    id: tableId,
    isDisabled: isLoading,
  };
};

export const filterConfigBuilder = (
  isLoading: boolean,
  setFilterValue: (value: any) => void = () => undefined,
  fetchData: (params: any) => Promise<any> | void = () => undefined,
  filterValue: string | string[] = '',
  pagination: PaginationData = defaultSettings,
  titleSingular = '',
  filterPlaceholder?: string,
  filterItems?: any[],
  filters?: FilterItem[],
  isFilterable?: boolean,
  onShowMore?: () => void,
  showMoreTitle?: string,
  onFilter?: (value: any) => void,
  onChange?: (event: any, value: any) => void,
  value?: any,
  sortBy?: string,
  textFilterRef?: RefObject<HTMLInputElement>,
): FilterConfig => {
  // TODO: PatternFly PrimaryToolbar sets generic "Options menu" aria-label for filter dropdowns
  // This is an accessibility issue - screen readers should hear "Filter by Status" not "Options menu"
  // Requires investigation into PatternFly's proper API for custom aria-labels
  const intl = useIntl();
  return {
    onChange,
    value,
    items: [
      ...(filters && filters.length > 0
        ? filters.map(({ key, label, value, selected, placeholder, type = 'text', groups, items, innerRef }) => ({
            label: label || firstUpperCase(key),
            type: type ?? 'text',
            filterValues: {
              innerRef,
              id: `filter-by-${key}`,
              key: `filter-by-${key}`,
              placeholder: placeholder ? placeholder : intl.formatMessage(messages.filterByKey, { key }),
              value,
              selected,
              ...(type !== 'text' ? { isFilterable, onShowMore, showMoreTitle, onFilter } : ({} as any)),
              groups,
              items,
              onChange: (_e: any, filterBy: any) => {
                const newFilter =
                  typeof filterBy !== 'string' && !Array.isArray(filterBy) ? Object.keys(pickBy(filterBy[''], (value) => value)) : filterBy;

                setFilterValue({
                  ...(typeof filterValue === 'string' ? { name: filterValue } : filterValue),
                  ...pagination,
                  offset: 0,
                  [key]: newFilter,
                });
                debouncedFetch(() =>
                  fetchData({
                    ...pagination,
                    offset: 0,
                    orderBy: sortBy,
                    ...filters.reduce(
                      (acc, curr) => ({
                        ...acc,
                        [curr.key]: curr.value,
                      }),
                      {},
                    ),
                    [key]: newFilter,
                  }),
                ).then((data) => {
                  innerRef?.current?.focus();
                  return data;
                });
              },
              isDisabled: isLoading,
            },
          }))
        : [
            {
              label: firstUpperCase(filterPlaceholder || titleSingular),
              type: 'text',
              filterValues: {
                innerRef: textFilterRef,
                id: 'filter-by-string',
                key: 'filter-by-string',
                placeholder: intl.formatMessage(messages.filterByKey, { key: filterPlaceholder || titleSingular }),
                value: filterValue,
                onChange: (_e: any, value: string) => {
                  setFilterValue({
                    ...pagination,
                    offset: 0,
                    name: value,
                  });
                  debouncedFetch(() =>
                    fetchData({
                      ...pagination,
                      offset: 0,
                      name: value,
                      orderBy: sortBy,
                    }),
                  ).then((data) => {
                    textFilterRef?.current?.focus();
                    return data;
                  });
                },
                isDisabled: isLoading,
              },
            },
          ]),
      ...(filterItems || []),
    ],
  };
};

export const activeFiltersConfigBuilder = (
  filterValue: string | string[] = '',
  filters?: FilterItem[],
  pagination: PaginationData = defaultSettings,
  setFilterValue: (value: any) => void = () => undefined,
  fetchData: (params: any) => Promise<any> | void = () => undefined,
  sortBy?: string,
): ActiveFiltersConfig => ({
  filters:
    filters && filters.length > 0
      ? filters
          .map(
            ({ key: type, value: options }) =>
              options &&
              (Array.isArray(options) || typeof options === 'string') &&
              options.length !== 0 && {
                category: firstUpperCase(type),
                type,
                chips: Array.isArray(options) ? options.map((filter) => ({ name: filter })) : [{ name: options }],
              },
          )
          .filter(Boolean)
      : [
          {
            name: typeof filterValue === 'string' ? filterValue : filterValue.join(', '),
          },
        ],
  onDelete: (_e: any, [deleted]: any[], isAll?: boolean) => {
    const setKeyValue = (value: any, type: string, key: string) => {
      if (isAll) {
        return type === 'group' || type === 'checkbox' ? [] : '';
      }

      if (key !== deleted.type) {
        return value;
      }

      if (type === 'checkbox' || type === 'group') {
        return value.filter((value: any) => value !== deleted.chips[0]?.name);
      }

      return Array.isArray(value) ? [] : '';
    };

    const filtersValue = filters?.reduce((acc, { key, value, type }) => ({ ...acc, [key]: setKeyValue(value, type || 'text', key) }), {}) || {};
    setFilterValue({
      ...pagination,
      offset: 0,
      name: '',
      ...filtersValue,
    });
    fetchData({
      ...pagination,
      offset: 0,
      orderBy: sortBy,
      name: '',
      ...filtersValue,
    });
  },
});

const Toolbar: React.FC<ToolbarProps> = ({
  isSelectable = false,
  isRowSelectable = () => true,
  checkedRows = [],
  setCheckedItems = () => undefined,
  isLoading = false,
  data = [],
  titleSingular = '',
  filterValue = [],
  setFilterValue = () => undefined,
  pagination = defaultSettings,
  paginationProps,
  fetchData,
  sortBy,
  toolbarButtons = () => [],
  filterPlaceholder,
  filterItems = [],
  filters = [],
  isFilterable = false,
  onShowMore,
  showMoreTitle,
  onFilter,
  onChange,
  value,
  hideFilterChips = false,
  tableId,
  textFilterRef,
  toolbarChildren = () => null,
}) => (
  <PrimaryToolbar
    {...(isSelectable && {
      bulkSelect: bulkSelectBuilder(isLoading, checkedRows, setCheckedItems, data, tableId, isRowSelectable),
    })}
    filterConfig={filterConfigBuilder(
      isLoading,
      setFilterValue,
      fetchData,
      filterValue,
      pagination,
      titleSingular,
      filterPlaceholder,
      filterItems,
      filters,
      isFilterable,
      onShowMore,
      showMoreTitle,
      onFilter,
      onChange,
      value,
      sortBy,
      textFilterRef,
    )}
    actionsConfig={{
      actions: toolbarButtons(),
    }}
    {...(!isLoading && {
      pagination: paginationBuilder(pagination, fetchData, filterValue, sortBy, paginationProps),
    })}
    {...(((typeof filterValue === 'string' ? filterValue.length > 0 : filterValue.length > 0) || (filters && filters.length > 0)) &&
      !hideFilterChips && {
        activeFiltersConfig: activeFiltersConfigBuilder(filterValue, filters, pagination, setFilterValue, fetchData, sortBy),
      })}
  >
    {toolbarChildren()}
  </PrimaryToolbar>
);

export { Toolbar };
