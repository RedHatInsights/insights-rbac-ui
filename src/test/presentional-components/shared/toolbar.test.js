import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Toolbar, {
  paginationBuilder,
  bulkSelectBuilder,
  filterConfigBuilder,
  activeFiltersConfigBuilder,
} from '../../../presentational-components/shared/toolbar';

const testPagination = {
  itemCount: undefined,
  page: 1,
  perPage: undefined,
  perPageOptions: [
    { title: '5', value: 5 },
    { title: '10', value: 10 },
    { title: '20', value: 20 },
    { title: '50', value: 50 },
  ],
};

const testBulkSelect = {
  checked: false,
  count: undefined,
  items: [{ title: 'Select none (0)' }, {}],
};

const testFilter = {
  items: [
    {
      filterValues: {
        id: 'filter-by-string',
        isDisabled: undefined,
        key: 'filter-by-string',
        placeholder: 'Filter by ',
        value: '',
      },
      label: '',
      type: 'text',
    },
  ],
};

const testActiveFilter = {
  filters: [{ name: '' }],
};

describe('<Toolbar>', () => {
  it('should render correctly - NO DATA', () => {
    const wrapper = shallow(<Toolbar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('isSelectable', () => {
    [true, false].map((isLoading) => {
      it(`is loading - ${isLoading}`, () => {
        const wrapper = shallow(<Toolbar isLoading={isLoading} isSelectable />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });

  describe('checkedRows', () => {
    [true, false].map((isLoading) => {
      it(`is loading - ${isLoading} NO DATA`, () => {
        const wrapper = shallow(<Toolbar isLoading={isLoading} isSelectable checkedRows={[{ uuid: 'some' }]} />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });

      it(`is loading - ${isLoading}`, () => {
        const wrapper = shallow(<Toolbar isLoading={isLoading} isSelectable checkedRows={[{ uuid: 'some' }]} data={[{ uuid: 'some' }]} />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });

  it('should render with filterValue', () => {
    const wrapper = shallow(<Toolbar filterValue="some" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render buttons', () => {
    const wrapper = shallow(
      <Toolbar
        toolbarButtons={() => [
          <button key={1}>Something</button>,
          {
            title: 'fff',
          },
        ]}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render pagination', () => {
    const wrapper = shallow(
      <Toolbar
        pagination={{
          count: 10,
          limit: 10,
          offset: 0,
        }}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  [true, false].map((isLoading) => {
    it(`should render with full data is loading - ${isLoading}`, () => {
      const wrapper = shallow(
        <Toolbar
          isLoading={false}
          isSelectable
          checkedRows={[{ uuid: 'some' }]}
          data={[{ uuid: 'some' }]}
          filterValue="some"
          pagination={{
            count: 10,
            limit: 10,
            offset: 0,
          }}
          toolbarButtons={() => [
            <button key={1}>Something</button>,
            {
              title: 'fff',
            },
          ]}
        />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('filters', () => {
    it('should render with placeholder', () => {
      const wrapper = shallow(<Toolbar filterValue="some" filterPlaceholder="test" />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with text filters', () => {
      const wrapper = shallow(<Toolbar filterValue="some" filters={[{ key: 'name', value: '' }]} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('call filter update with correct value', () => {
      const setFilterValue = jest.fn();
      const wrapper = mount(<Toolbar filterValue="some" filters={[{ key: 'name', value: '' }]} setFilterValue={setFilterValue} />);
      const target = wrapper.find('input#filter-by-name');
      target.getDOMNode().value = 'something';
      target.simulate('change');
      wrapper.update();
      expect(setFilterValue).toHaveBeenCalled();
      expect(setFilterValue.mock.calls[0][0]).toMatchObject({ name: 'something' });
    });
  });
});

describe('paginationBuilder', () => {
  it('should return correct config - NO DATA', () => {
    const config = paginationBuilder();
    expect(config).toMatchObject(testPagination);
  });
});

describe('bulkSelectBuilder', () => {
  it('should return correct config - NO DATA', () => {
    const config = bulkSelectBuilder();
    expect(config).toMatchObject(testBulkSelect);
  });
});

describe('filterConfigBuilder', () => {
  it('should return correct config - NO DATA', () => {
    const config = filterConfigBuilder();
    expect(config).toMatchObject(testFilter);
  });
});

describe('activeFiltersConfigBuilder', () => {
  it('should return correct config - NO DATA', () => {
    const config = activeFiltersConfigBuilder();
    expect(config).toMatchObject(testActiveFilter);
  });
});
