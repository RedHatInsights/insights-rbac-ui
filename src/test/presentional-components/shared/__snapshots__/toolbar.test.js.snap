// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar> checkedRows is loading - false 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [],
    }
  }
  bulkSelect={
    Object {
      "checked": true,
      "count": 1,
      "items": Array [
        Object {
          "onClick": [Function],
          "title": "Select none (0)",
        },
        Object {
          "onClick": [Function],
          "title": "Select page (1)",
        },
      ],
      "onSelect": [Function],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": false,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": Array [],
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  pagination={
    Object {
      "itemCount": undefined,
      "limit": 20,
      "offset": 0,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 20,
      "perPageOptions": Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ],
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> checkedRows is loading - false NO DATA 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [],
    }
  }
  bulkSelect={
    Object {
      "checked": false,
      "count": 1,
      "items": Array [
        Object {
          "onClick": [Function],
          "title": "Select none (0)",
        },
        Object {},
      ],
      "onSelect": [Function],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": false,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": Array [],
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  pagination={
    Object {
      "itemCount": undefined,
      "limit": 20,
      "offset": 0,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 20,
      "perPageOptions": Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ],
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> checkedRows is loading - true 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [],
    }
  }
  bulkSelect={
    Object {
      "checked": true,
      "count": 1,
      "items": Array [
        Object {
          "onClick": [Function],
          "title": "Select none (0)",
        },
        Object {},
      ],
      "onSelect": [Function],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": true,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": Array [],
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> checkedRows is loading - true NO DATA 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [],
    }
  }
  bulkSelect={
    Object {
      "checked": false,
      "count": 1,
      "items": Array [
        Object {
          "onClick": [Function],
          "title": "Select none (0)",
        },
        Object {},
      ],
      "onSelect": [Function],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": true,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": Array [],
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> filters should render with placeholder 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [],
    }
  }
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "name": "some",
        },
      ],
      "onDelete": [Function],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": false,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by test",
            "value": "some",
          },
          "label": "Test",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  pagination={
    Object {
      "itemCount": undefined,
      "limit": 20,
      "offset": 0,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 20,
      "perPageOptions": Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ],
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> filters should render with text filters 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [],
    }
  }
  activeFiltersConfig={
    Object {
      "filters": Array [],
      "onDelete": [Function],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "groups": undefined,
            "id": "filter-by-name",
            "isDisabled": false,
            "isFilterable": false,
            "items": undefined,
            "key": "filter-by-name",
            "onChange": [Function],
            "onFilter": undefined,
            "onShowMore": undefined,
            "placeholder": "Filter by name",
            "selected": undefined,
            "showMoreTitle": undefined,
            "value": "",
          },
          "label": "Name",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  pagination={
    Object {
      "itemCount": undefined,
      "limit": 20,
      "offset": 0,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 20,
      "perPageOptions": Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ],
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> isSelectable is loading - false 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [],
    }
  }
  bulkSelect={
    Object {
      "checked": false,
      "count": undefined,
      "items": Array [
        Object {
          "onClick": [Function],
          "title": "Select none (0)",
        },
        Object {},
      ],
      "onSelect": [Function],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": false,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": Array [],
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  pagination={
    Object {
      "itemCount": undefined,
      "limit": 20,
      "offset": 0,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 20,
      "perPageOptions": Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ],
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> isSelectable is loading - true 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [],
    }
  }
  bulkSelect={
    Object {
      "checked": false,
      "count": undefined,
      "items": Array [
        Object {
          "onClick": [Function],
          "title": "Select none (0)",
        },
        Object {},
      ],
      "onSelect": [Function],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": true,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": Array [],
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> should render buttons 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [
        <button>
          Something
        </button>,
        Object {
          "title": "fff",
        },
      ],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": false,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": Array [],
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  pagination={
    Object {
      "itemCount": undefined,
      "limit": 20,
      "offset": 0,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 20,
      "perPageOptions": Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ],
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> should render correctly - NO DATA 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": false,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": Array [],
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  pagination={
    Object {
      "itemCount": undefined,
      "limit": 20,
      "offset": 0,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 20,
      "perPageOptions": Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ],
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> should render pagination 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": false,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": Array [],
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  pagination={
    Object {
      "count": 10,
      "itemCount": 10,
      "limit": 10,
      "offset": 0,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 10,
      "perPageOptions": Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ],
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> should render with filterValue 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [],
    }
  }
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "name": "some",
        },
      ],
      "onDelete": [Function],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": false,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": "some",
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  pagination={
    Object {
      "itemCount": undefined,
      "limit": 20,
      "offset": 0,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 20,
      "perPageOptions": Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ],
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> should render with full data is loading - false 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [
        <button>
          Something
        </button>,
        Object {
          "title": "fff",
        },
      ],
    }
  }
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "name": "some",
        },
      ],
      "onDelete": [Function],
    }
  }
  bulkSelect={
    Object {
      "checked": true,
      "count": 1,
      "items": Array [
        Object {
          "onClick": [Function],
          "title": "Select none (0)",
        },
        Object {
          "onClick": [Function],
          "title": "Select page (1)",
        },
      ],
      "onSelect": [Function],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": false,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": "some",
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  pagination={
    Object {
      "count": 10,
      "itemCount": 10,
      "limit": 10,
      "offset": 0,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 10,
      "perPageOptions": Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ],
    }
  }
  toggleIsExpanded={[Function]}
/>
`;

exports[`<Toolbar> should render with full data is loading - true 1`] = `
<PrimaryToolbar
  actionsConfig={
    Object {
      "actions": Array [
        <button>
          Something
        </button>,
        Object {
          "title": "fff",
        },
      ],
    }
  }
  activeFiltersConfig={
    Object {
      "filters": Array [
        Object {
          "name": "some",
        },
      ],
      "onDelete": [Function],
    }
  }
  bulkSelect={
    Object {
      "checked": true,
      "count": 1,
      "items": Array [
        Object {
          "onClick": [Function],
          "title": "Select none (0)",
        },
        Object {
          "onClick": [Function],
          "title": "Select page (1)",
        },
      ],
      "onSelect": [Function],
    }
  }
  filterConfig={
    Object {
      "items": Array [
        Object {
          "filterValues": Object {
            "id": "filter-by-string",
            "isDisabled": false,
            "key": "filter-by-string",
            "onChange": [Function],
            "placeholder": "Filter by ",
            "value": "some",
          },
          "label": "",
          "type": "text",
        },
      ],
      "onChange": undefined,
      "value": undefined,
    }
  }
  pagination={
    Object {
      "count": 10,
      "itemCount": 10,
      "limit": 10,
      "offset": 0,
      "onPerPageSelect": [Function],
      "onSetPage": [Function],
      "page": 1,
      "perPage": 10,
      "perPageOptions": Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ],
    }
  }
  toggleIsExpanded={[Function]}
/>
`;
